### PR TITLE
XIVY-12538 program 'userconfig' as map

### PR DESCRIPTION
**Pull request by ivy-rew**

### CODE REVIEW

**Review by ivy-lli — APPROVED** [review]

File: packages/editor/src/components/parts/program/configuration/ConfigurationPart.tsx [hunk 1]
```diff
@@ -76,7 +76,7 @@ const ConfigurationPart = () => {
   return (
     <>
       {editorItems.map((widget, index) => (
-        <div key={index}>{renderWidgetComponent(widget)}</div>
+        <div key={index}>{renderWidgetComponent(widget, index.toString())}</div>
```
Comment (ivy-lli) [inline packages/editor/src/components/parts/program/configuration/ConfigurationPart.tsx:79]: not sure that this is correct. 
As far as I understood this, you should now have a id on the widget, which is corresponding to the key of the config data structure

File: packages/protocol/src/defaults/common-defaults.ts [hunk 2]
```diff
@@ -231,6 +232,11 @@ export const DEFAULT_PROGRAM_INTERFACE_START_DATA: ProgramInterfaceStartData = {
   }
 } as const;
 
+export const DEFAULT_CONFIGURATION_DATA: ConfigurationData = {
+  javaClass: '',
+  userConfig: { '1': '' }
```
Comment (ivy-lli) [inline packages/protocol/src/defaults/common-defaults.ts:237]: i think the default of this is an empty object...

**Review by ivy-lli — CHANGES_REQUESTED** [review]

File: packages/editor/src/components/parts/program/configuration/ConfigurationPart.test.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ describe('ConfigurationPart', () => {
 
   test('full data', async () => {
     renderPart({
-      userConfig: '123'
+      userConfig: { directory: '123' }
```
Comment (ivy-rew) [inline packages/editor/src/components/parts/program/configuration/ConfigurationPart.test.tsx:34]: directory : /tmp/myDir

File: packages/protocol/src/defaults/common-defaults.ts [hunk 4]
```diff
@@ -234,7 +234,7 @@ export const DEFAULT_PROGRAM_INTERFACE_START_DATA: ProgramInterfaceStartData = {
 
 export const DEFAULT_CONFIGURATION_DATA: ConfigurationData = {
   javaClass: '',
-  userConfig: { '1': '' }
+  userConfig: { '': '' }
```
Comment (ivy-lli) [inline packages/protocol/src/defaults/common-defaults.ts:237]: why not {}?

**Review by ivy-lli — APPROVED** [review]

**Pull request merged**